lib/tasks/existing_firms_sign_up_task.rb

```diff
@@ -30,8 +22,17 @@ def self.invite(principal)
       )
     end
 
-    def self.valid_registered_parent_firms
-      Firm.registered.all.select { |f| f.valid? && f.parent.nil? }
+    def self.build_csv_data(inviter, user)
+      principal = user.principal
+      firm = principal.firm
+
+      result = [firm.fca_number]
+      result << firm.registered_name
+      result << principal.full_name
+      result << principal.email_address
+      result << inviter.invitation_url(user)
+      result << (firm.email_address.present? ? 'registered' : 'not registered')
+      result
```

Comment:
This could be put more concisely like:

```ruby
[
  firm.fca_number,
  firm.registered_name,
  principal.full_name,
  principal.email_address,
  inviter.invitation_url(user),
  (firm.email_address.present? ? 'registered' : 'not registered')
]
```

your call